lighthouse-core/gather/gatherers/scripts.js

```diff
@@ -137,15 +145,27 @@ class Scripts extends FRGatherer {
       // It's nice to display the user-provided value in Lighthouse, so we add a field 'name'
       // to make it clear this is for presentational purposes.
       // See https://chromium-review.googlesource.com/c/v8/v8/+/2317310
+      let name = event.url;
+      // embedderName is optional on the protocol because backends like Node may not set it.
+      // For our purposes, it is always set. But just in case it isn't... fallback to the url.
+      let url = event.embedderName || event.url;
+
+      // Some eval'd scripts may not have a name or url, so let's set them.
+      if (!name) name = '<eval script>';
+      if (!url && event.stackTrace?.callFrames.length) url = event.stackTrace.callFrames[0].url;
+      if (!url) url = this._scriptFrameUrls[i] || '';
+
       return {
-        name: event.url,
+        name,
         ...event,
-        // embedderName is optional on the protocol because backends like Node may not set it.
-        // For our purposes, it is always set. But just in case it isn't... fallback to the url.
-        url: event.embedderName || event.url,
+        url,
         content: this._scriptContents[i],
       };
-    });
+    })
+    // If we can't name a script or know its url, just ignore it.
+    .filter(script => script.name && script.url)
+    // This is a script with unknown origin, but seems to come from Puppeteer.
+    .filter(script => script.content !== '(function(e) { console.log(e.type, e); })');
```

Comment:
This only happens in FR mode, hence I suspected Puppeteer. But I instrumented every call to `Runtime.evaluate` in Puppeteer and did not find it...

Reply:
ah, well I guess you can do `someFn = eval('(function() {})')`

Reply:
> I saw it happening in legacy nav as well on [paulirish.com](https://paulirish.com/)

yea me too

Reply:
https://source.chromium.org/chromium/chromium/src/+/main:third_party/blink/renderer/core/inspector/thread_debugger.cc?q=%22(function(e)%20%7B%20console.log(e.type,%20e);%20%7D)%22&ss=chromium